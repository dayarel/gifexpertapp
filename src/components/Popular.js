import React from "react";
import PropTypes from "prop-types";

const Popular = ({ setCategories }) => {
  return (
    <div className="popular">
      <h3>Popular</h3>
      <div className="popular-wrapper">
        <button
          className="boton"
          onClick={() => setCategories((cats) => ["Overwatch", ...cats])}
        >
          Overwatch
        </button>
        <button onClick={() => setCategories((cats) => ["Capitol", ...cats])}>
          Capitol
        </button>
        <button onClick={() => setCategories((cats) => ["Pokemon", ...cats])}>
          Pokemon
        </button>
        <button onClick={() => setCategories((cats) => ["Cats", ...cats])}>
          Cats
        </button>
        <button onClick={() => setCategories((cats) => ["Covid", ...cats])}>
          Covid
        </button>
        <button
          onClick={() => setCategories((cats) => ["Animal Crossing", ...cats])}
        >
          Animal Crossing
        </button>
      </div>
    </div>
  );
};

Popular.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
export default Popular;
