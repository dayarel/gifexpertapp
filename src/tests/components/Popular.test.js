import React from "react";
import Popular from "../../components/Popular";
import { shallow } from "enzyme";

describe("Tests sobre Popular", () => {
  const setCategories = jest.fn();
  const wrapper = shallow(<Popular setCategories={setCategories} />);

  test("se muestra el componente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Se cambia la categoria cuando se pulsa el botn", () => {
    const boton = wrapper.find(".boton");

    boton.simulate("click");

    expect(setCategories).toHaveBeenCalled();
  });
});
