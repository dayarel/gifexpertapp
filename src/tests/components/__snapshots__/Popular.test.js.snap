// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests sobre Popular se muestra el componente 1`] = `
<div
  className="popular"
>
  <h3>
    Popular
  </h3>
  <div
    className="popular-wrapper"
  >
    <button
      className="boton"
      onClick={[Function]}
    >
      Overwatch
    </button>
    <button
      onClick={[Function]}
    >
      Capitol
    </button>
    <button
      onClick={[Function]}
    >
      Pokemon
    </button>
    <button
      onClick={[Function]}
    >
      Cats
    </button>
    <button
      onClick={[Function]}
    >
      Covid
    </button>
    <button
      onClick={[Function]}
    >
      Animal Crossing
    </button>
  </div>
</div>
`;
